src/rubyfmt.rb

```diff
@@ -716,6 +716,25 @@ def format_int(ps, rest)
   ps.emit_indent if ps.start_of_line.last
 
   int = rest[0]
+
+  if int[0] != "0"
+    int_array = []
+    counter = 0
+
+    (0..int.length - 1).each do |i|
+      next if int[int.length - 1 - i] == "_"
```

Comment:
I’m not sure how important performance is yet, but I fiddled a bit with your code to reduce array lookups:
```rb
  if int[0] != "0"
    int_array = []
    counter = 0

    int.chars.reverse.each do |char|
      next if char == "_"

      if counter > 0 && counter % 3 == 0
        int_array.unshift("_")
      end

      int_array.unshift(char)
      counter += 1
    end

    int = int_array.join
  end
```
Simple benchmarking suggests the fixtures file is handled around 45% faster, and exclusively working on small integers works 15% faster.

Reply:
Nice suggestion -- I benchmarked against some options (including `push` vs `unshift`) initially, but it definitely seems like this is a good improvement.